public_html/foods.php

```diff
@@ -84,21 +84,69 @@
         </form>
 <?php   break;
         case 'history':
-            if(isset($_GET['timestamp'])):
-                echo "        <h3 class=\"mt-3\"><a href=\"?action=history\">Your meal history</a> &rsaquo; Your meal on " . date('h:i:s A l, jS \of F Y', (int)$_GET['timestamp']) . "</h3>\n";
-                $meals = $db->get_meals((int)$_SESSION['user_id'], (int)$_GET['timestamp']);
-                foreach($meals as $meal):
-                    draw_table($meal['foods'], ['name' => 'Food', 'calories' => 'Calories']);
-                endforeach;
-            else:
-                echo "        <h3 class=\"mt-3\">Your meal history</h2>\n";
-                $meals = $db->get_meals((int)$_SESSION['user_id']);
-                foreach($meals as $meal):
-                    echo "\n            <h5 class=\"mt-3\"><a href=\"?action=history&timestamp=" . $meal['unixtime'] . "\">". date('h:i:s A l, jS \of F Y', $meal['unixtime']) . "</a></h3>\n";
-                    draw_table($meal['foods'], ['name' => 'Food', 'calories' => 'Calories']);
-                endforeach;
-            endif;
-        break;
+?>
+<?php
+        /**
+         * Sort order of a table
+         * 
+         * This function takes a sort variable and returns the url of the sort type.
+         * @return url
+         * Appends to the end of the url.
+         */
+        function sortorder($fieldname) {
+            $sorturl = "&orderBy=".$fieldname."&sort=";
+            $sorttype = "asc";
+
+            if(isset($_GET['orderBy']) && $_GET['orderBy'] == $fieldname){
+                if(isset($_GET['sort']) && $_GET['sort'] == "asc"){
+                    $sorttype = "desc";
+                }
+            }
+            $sorturl .= $sorttype;
+            return $sorturl;
+        } 
```

Comment:
This function is really cool. Is it possible to include it in `library.php` so we can reuse it in other places?